bvm/ballerina-core/src/main/java/org/ballerinalang/util/TableProvider.java

```diff
@@ -0,0 +1,328 @@
+                        byte[] blobData = constrainedType.getBlobField(blobFieldIndex);
+                        stmt.setBlob(index, new ByteArrayInputStream(blobData), blobData.length);
+                        ++blobFieldIndex;
+                        break;
+                    case TypeTags.ARRAY_TAG:
+                        Object[] arrayData = getArrayData(constrainedType.getRefField(refFieldIndex));
+                        stmt.setObject(index, arrayData);
+                        ++refFieldIndex;
+                        break;
+                    }
+                ++index;
+            }
+            stmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new BallerinaException(
+                    "error in executing statement : " + queryStatement + " error:" + e.getMessage());
+        } finally {
+            releaseResources(conn, stmt);
+        }
+    }
+
+    private static Object[] getArrayData(BValue value) {
+        if (value == null) {
+            return new Object[] {null};
+        }
+        int typeTag = ((BArrayType) value.getType()).getElementType().getTag();
+        Object[] arrayData;
+        int arrayLength;
+        switch (typeTag) {
+        case TypeTags.INT_TAG:
+            arrayLength = (int) ((BIntArray) value).size();
+            arrayData = new Long[arrayLength];
+            for (int i = 0; i < arrayLength; i++) {
+                arrayData[i] = ((BIntArray) value).get(i);
+            }
+            break;
+        case TypeTags.FLOAT_TAG:
+            arrayLength = (int) ((BFloatArray) value).size();
+            arrayData = new Double[arrayLength];
+            for (int i = 0; i < arrayLength; i++) {
+                arrayData[i] = ((BFloatArray) value).get(i);
+            }
+            break;
+        case TypeTags.STRING_TAG:
+            arrayLength = (int) ((BStringArray) value).size();
+            arrayData = new String[arrayLength];
+            for (int i = 0; i < arrayLength; i++) {
+                arrayData[i] = ((BStringArray) value).get(i);
+            }
+            break;
+        case TypeTags.BOOLEAN_TAG:
+            arrayLength = (int) ((BBooleanArray) value).size();
+            arrayData = new Boolean[arrayLength];
+            for (int i = 0; i < arrayLength; i++) {
+                arrayData[i] = ((BBooleanArray) value).get(i) > 0;
+            }
+            break;
+        case TypeTags.BLOB_TAG:
+            arrayLength = (int) ((BBlobArray) value).size();
+            arrayData = new Blob[arrayLength];
+            for (int i = 0; i < arrayLength; i++) {
+                arrayData[i] = ((BBlobArray) value).get(i);
+            }
+            break;
+        default:
+            throw new BallerinaException("unsupported data type for array parameter");
+        }
+        return arrayData;
+    }
+
+    private void releaseResources(Connection conn, Statement stmt) {
+        try {
+            if (stmt != null) {
```

Comment:
Better to use two try catch statements, otherwise, connection may not get closed(if stmt.close() fails)

Reply:
Fixed with 4aa2f2bde65e1723c1ebe3fe84770279a6d7bf46